sell-loot.lic

```diff
@@ -80,15 +88,15 @@ class SellLoot
       .each { |currency| fput "exchange all #{currency} for #{exchange_to}" }
   end
 
-  def deposit_coins(keep_copper)
+  def deposit_coins(keep_money_by_currency)
     walk_to @hometown['deposit']['id']
     release_invisibility
     bput('wealth', 'Wealth:')
     case bput('deposit all', 'you drop all your', 'You hand the clerk some coins', "You don't have any", 'There is no teller here', 'reached the maximum balance I can permit')
     when 'There is no teller here'
       return
     end
-    minimize_coins(keep_copper).each { |amount| withdraw_exact_amount?(amount, @settings) }
+    fput("withdraw #{keep_money_by_currency}")
```

Comment:
❓ This doesn't look equivalent. Why did this change?

Reply:
It's equivalent due to the 'deposit all' on line 95.  There are no coins to minimize at this point, and withdraw_exact_amount runs to your hometown instead of the bank you're already in

Reply:
Good question.  I can't think of anything, but don't know what edge cases there could be.

Reply:
You may want to branch on the town arg here, as the above two methods seem to handle bankbot transactions, being invisible, withdrawing too quickly, and not having enough money. fput would cover the last two but not the first two

Reply:
e.g. 
if args.town -> fput
else -> minimize+ensure

Reply:
Bankbots get handled on line 65.  This method is only called if no bankbot is set, and line 93 includes the `release_invisibility` check

Reply:
Sorry, bankbots get handled on line 68

Reply:
`withdraw_exact_amount?` withdraws from bankbots and releases invisibility

Reply:
:bulb: im just pointing out that this is going to remove those two things. I dont know who uses them or needs them. The whole banking thing probably needs an overhaul. things seem too tightly coupled (like movement is tied to withdrawing as you pointed out)

Reply:
Right, but you won't be in the bank depositing if you've got a bank bot or invisible.  Those checks have already been made.

Agree on cleaning up some of the banking code.  Maybe that makes more sense to tackle first, but that would be a pretty big task

Reply:
I think branching on your feature, e.g. args.town would still work though? Else just do business as usual

Reply:
I disagree - I added the invisibility check because I train with that as a cyclic spell and it was breaking other things. Id end up stuck at the bank because i cant withdraw. I really think theres an easy solution to this without removing existing functionality.

Reply:
Sorry, I'm probably coming across too strong, didn't intend for that. I just wanted to point out some code that might be affected. Carry on :)